### PR TITLE
Validate lammps run

### PR DESCRIPTION
**Pull request by sudarsan-surendralal**
Check if atoms are within box (along non-periodic sides)

### CODE REVIEW

**Comment by samwaseda** [comment]
I have the feeling that this could occur due to python numerical precision (i.e. something like -1.0e-36 was set instead of 0). Wouldn't it be better if we replace the error by a warning and we perform `center_coordinates_in_unit_cell` behind the scenes? But I'd follow what everyone prefers.

In this regard I add @liamhuber

**Comment by sudarsan-surendralal** [comment]
> 
> 
> I have the feeling that this could occur due to python numerical precision (i.e. something like -1.0e-36 was set instead of 0). Wouldn't it be better if we replace the error by a warning and we perform `center_coordinates_in_unit_cell` behind the scenes? But I'd follow what everyone prefers.
> 
> In this regard I add @liamhuber

I agree with the python precision part however, I don't think we should center the coordinates for a structure without pbc since it does not make sense physically (esp. for molecules).

**Comment by samwaseda** [comment]
Oh sorry I didn't see that pbc was also checked. Ok then I'm fine.

File: pyiron/lammps/base.py [hunk 1]
```diff
@@ -212,15 +212,20 @@ def set_input_to_read_only(self):
...
-        Returns:
-
+        Validating input parameters before LAMMPS run
         """
         super(LammpsBase, self).validate_ready_to_run()
         if self.potential is None:
             raise ValueError(
                 "This job does not contain a valid potential: {}".format(self.job_name)
             )
+        scaled_positions = self.structure.get_scaled_positions()
```
Comment (samwaseda) [inline pyiron/lammps/base.py:222]: `get_scaled_positions(wrap=False)`

Comment (samwaseda) [reply pyiron/lammps/base.py:222]: I mean, it's unlikely that this is required, but I guess it's safer.

Comment (sudarsan-surendralal) [reply pyiron/lammps/base.py]: Done!

**Pull request merged**